docs/admin/kubeadm-init.md

```diff
@@ -0,0 +1,425 @@
+* `--cert-dir`the directory where the TLS certs are located.
+
+### Use kubeadm with other CRI runtimes
+
+Since the [Kubernetes 1.6 release](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG.md#node-components-1),
+Kubernetes has started using CRI as the default container runtime..
+the container runtime used by kubeadm is Docker, which is enabled through the built-in
+`dockershim` in `kubelet` mechanisms.
+
+Other CRI-based runtimes include::
+
+- [cri-o](https://github.com/kubernetes-incubator/cri-o)
+- [frakti](https://github.com/kubernetes/frakti)
+- [rkt](https://github.com/kubernetes-incubator/rktlet)
+
+After you have successfully installed `kubeadm` and `kubelet`, please follow
+these two steps:
+
+1. Install runtime shim on every node. You will need to follow the installation
+   document in the runtime shim project listing above.
+
+1. Configure kubelet to use remote CRI runtime. Please remember to change
+   `RUNTIME_ENDPOINT` to your own value like `/var/run/{your_runtime}.sock`:
+
+```shell
+  $ cat > /etc/systemd/system/kubelet.service.d/20-cri.conf <<EOF
+Environment="KUBELET_EXTRA_ARGS=--container-runtime=remote --container-runtime-endpoint=$RUNTIME_ENDPOINT --feature-gates=AllAlpha=true"
+EOF
+  $ systemctl daemon-reload
+```
+
+Now `kubelet` is ready to use the specified CRI runtime, and you can continue
+with `kubeadm init` and `kubeadm join` workflow to deploy Kubernetes cluster.
+
+### Using internal IPs in your cluster
+
+In order to set up a cluster where the master and worker nodes communicate with internal IP addresses (instead of publicly addressable ones), you will need to follow these steps.
+
+1. When running init, you must make sure you specify an internal IP for the apiserver's bind address, like so:
+
+   `kubeadm init --apiserver-advertise-address=<private-master-ip>`
+
+2. When a worker node has been provisioned, add a flag to `/etc/systemd/system/kubelet.service.d/10-kubeadm.conf` that specifes the private IP of the worker node:
+
+   `--node-ip=<private-node-ip>`
+
+3. Finally, when you run `kubeadm join` make sure you provide the private IP of the apiserver addressed as defined in step 1.
+
+### Self-hosting the Kubernetes control plane  {#self-hosting}
+As of 1.8, kubeadm can experimentally create a _self-hosted_ Kubernetes control
+plane. This means that key components such as the API server, controller
+manager, and scheduler run as [DaemonSet pods](/docs/concepts/workloads/controllers/daemonset/)
+configured via the Kubernetes API instead of [static pods](/docs/tasks/administer-cluster/static-pod/)
+configured in the kubelet via static files.
+
+**Caution:** Self-hosting is alpha in kubeadm 1.8 but is expected to become the default in
+a future version. To create a self-hosted cluster, pass the `--feature-gates=SelfHosting=true`
+flag to `kubeadm init`.
+{: .caution}
+
+**Warning:** Read carefully current caveats/limitations. 
+{: .warning}
+
```

Comment:
Avoid saying "current".
Read carefully about caveats and limitations in kubeadm 1.8.